core/src/main/java/org/mule/processor/NonBlockingMessageProcessor.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.processor;
+
+import org.mule.api.processor.MessageProcessor;
+
+/**
+ * A {@link org.mule.processor.NonBlockingMessageProcessor} differs from other message processors in that it supports
+ * both blocking and non-blocking processing.
```

Comment:
The fact that NonBlockingMessageProcessor support blocking and non-blocking processing makes me think that maybe the name is wrong and it could imply that always executes in a non-blocking way